### PR TITLE
Add "warm welcome" for newly purchased plans

### LINKED ISSUE #7374: Welcome message post-purchase on Jetpack dashboard
This is another part of giving our users a warm welcome. We should show a view that gives them a bit of a smile and encourages them that what they just purchased is pretty dang cool/valuable. This should be implemented in the same fashion as used in [Jumpstart](https://github.com/Automattic/jetpack/issues/7373).

This is a really rough version of what it could look like. Obviously the copy could use a bit of a massage.

image

More about warm welcome: p6TEKc-1hL

### PR DESCRIPTION
**Pull request by dereksmart**
Fixes #7374

This PR works in the following technical ways:
 - If the Jetpack plan changes, an option `show_welcome_for_new_plan` will be set and act as a flag to whether or not a "new plan" was purchased. It should not set if the plan changes/downgrades to "free"
- New `<WelcomeNewPlan />` component, which acts as a "parent" component to the dummy individual welcomes.  
- Dismissing simply sets the flag option to false.

To Test: 
- Upgrade to any plan.  
- You should see the warm welcome on any admin screen
- Dismissing should dismiss
- Links should all dismiss and take you to the respective places
- Should not be able to see it at all if not an admin, or on free plan of any kind.

Hacks to make testing easier: 
- manually set the flag option `update_option( 'show_welcome_for_new_plan', true );` and then using the dev tools, select a paid plan.
- Since the modal is hiding the "dev tools" footer link, you can manually trigger it by writing `window.dispatch( { type: "DEV_CARD_DISPLAY" } );` in the browser console.  

In progress until @MichaelArestad gets his hands on it

### CODE REVIEW

**Comment by MichaelArestad** [comment]
Updated the content with help from @nicoleckohler.

Screenshots:

image
image
image

Sorry about the weird crops. My screenshotting tool stopped working :/ CC @rickybanister 

@dereksmart I think everything works, but it could use some double checking. Also, you might have a better idea how to insert/use the SVGs. They don't have to be inline.

**Comment by dereksmart** [comment]
@MichaelArestad thanks.  I pushed a30a4bf which cleans up the inline svgs.  We already had some of them in Jetpack :) 

I also renamed a couple of the new ones based on their ID.  Figured it would be easier to cross reference later.  

This is ready for final review.

**Comment by MichaelArestad** [comment]
Nice. Thank you, Derek!  Let's get it code reviewed and in. I'd like to swap one of the images in the Personal plan, but I can do that in a further PR.

File: _inc/client/state/settings/reducer.js [hunk 1]
```diff
@@ -201,3 +201,12 @@ export function appsCardDismissed( state ) {
 export function emptyStatsCardDismissed( state ) {
 	return get( state.jetpack.settings.items, 'dismiss_empty_stats_card', false );
 }
+
+/**
+ * Returns true if Empty Stats card has been dismissed.
```
Comment (oskosk) [inline _inc/client/state/settings/reducer.js]: This description for `showWelcomeForNewPlan ` is weird

**Comment by oskosk** [comment]
For the Premium plan, click the link "Montieze your site with ads", takes me to the traffic tab, but on a random scroll position. Well, this happens for every link taking to Traffic on Premium and Professional plans, so it's consistent.

Other than that, LGTM !

File: _inc/client/state/settings/reducer.js (hunk 1 above)
Comment (dereksmart) [reply _inc/client/state/settings/reducer.js]: bad copy pasta.  fixed in 6c5c299ae

**Comment by MichaelArestad** [comment]
@dereksmart are the buttons recording events in tracks?

**Comment by oskosk** [comment]
Don't know if this is relevant, but, by switching with the dev tools from admin, to editor, and then from free to premium, I still get to see the modal

**Comment by dereksmart** [comment]
@oskosk I wouldn't really trust the "role" option in the dev tool ;)  Best to log in as a real editor

**Comment by dereksmart** [comment]
@MichaelArestad I added tracking in f3f2858d6

@withinboredom could you give that commit a review?

**Comment by dereksmart** [comment]
@MichaelArestad PR incoming with tracks

**Pull request merged**